<a href="https://colab.research.google.com/github/AliyevaO/Loan/blob/main/Project_2_Part_1_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

* Oleksandra Aliyeva
* 03.12.2022

## Dataset 1




In [1]:
#mount google disk
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#import some necessary tools
from sklearn.model_selection import train_test_split
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.compose import make_column_transformer, make_column_selector
from sklearn.impute import SimpleImputer

#import models for ML
from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import  RandomForestClassifier

#import accuracy, precision, recall, classification report, and confusion matrix scoring functions
from sklearn.metrics import accuracy_score, classification_report, ConfusionMatrixDisplay
from sklearn.model_selection import GridSearchCV 
from sklearn import set_config
set_config(display='diagram')

1) Source of data:

https://www.kaggle.com/vikramkumar001/partial-bank-loan-dataset

In [3]:
loan_df = pd.read_csv('/content/drive/MyDrive/Bootcamp/Data/partial_loan_dataset.csv')
loan_df.head()

Loan_ID  Gender Married  Dependent_No     Education  Self_Employed  \
0    23002    Male    True             3      Graduate           True   
1    23003  Female    True             3  Not Graduate           True   
2    23004    Male    True             3  Not Graduate           True   
3    23005    Male    True             2  Not Graduate          False   
4    23006    Male   False             2  Not Graduate          False   

   Applicant_Income  CoApplicant_Income  Loan_Amount  Loan_Amount_Term  \
0              4315             19748.0        60000               300   
1              3331             29657.0        20000                60   
2             22494             29539.0        30000               240   
3             78117             12909.0        20000                60   
4             41976             45598.0        60000               300   

   Credit History Property_District  Loan_Status  
0               1               OCR            1  
1               0               OCR            1  
2               1               RCR            1  
3               0               OCR            1  
4               1               RCR            1

2) Brief description of data.

This Data set has information about appications for reciving a loan. 
Company wants to identify applicants that are eligible for the loan amount requested so that those who met the requirements will be granted loan and reject those who failed.
Using Machine Learning Models I will need to make a prediction if person is iligible for loan or not.

In [4]:
loan_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5584 entries, 0 to 5583
Data columns (total 13 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Loan_ID             5584 non-null   int64  
 1   Gender              5425 non-null   object 
 2   Married             5328 non-null   object 
 3   Dependent_No        5584 non-null   int64  
 4   Education           5218 non-null   object 
 5   Self_Employed       5584 non-null   bool   
 6   Applicant_Income    5584 non-null   int64  
 7   CoApplicant_Income  4999 non-null   float64
 8   Loan_Amount         5584 non-null   int64  
 9   Loan_Amount_Term    5584 non-null   int64  
 10  Credit History      5584 non-null   int64  
 11  Property_District   5578 non-null   object 
 12  Loan_Status         5584 non-null   int64  
dtypes: bool(1), float64(1), int64(7), object(4)
memory usage: 529.1+ KB


3) What is the target?

My target column will be - 'Loan_Status' 

4) Is this a classification or regression problem?

It is a classification problem. I decided to choose a set for classification, because I used regression in my first Project.

5) How many features? 

This Data has 13 columns.

6) How many rows of data?

This Data has 5584 rows.

In [5]:
loan_df.shape

(5584, 13)


7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

I think everything we did before will help me to do cleaning, the rest will not be very difficult for me because I'm familiar with loan aprovement proces. 

## Dataset 2

1) Source of data:

https://www.kaggle.com/nurulsyazanaramlee/diabetes-prediction

In [6]:
diabetes_df = pd.read_csv('/content/drive/MyDrive/Bootcamp/Data/diabetesprediction.csv.xls')
diabetes_df.head()

NoofPregnancies  GlucoseLevel  DiastolicPressure  Insulin Level   BMI  \
0                6           148                 72              0  33.6   
1                1            85                 66              0  26.6   
2                8           183                 64              0  23.3   
3                1            89                 66             94  28.1   
4                0           137                 40            168  43.1   

   RelativeHistory  Age  Diabetic  
0            0.627   50         1  
1            0.351   31         0  
2            0.672   32         1  
3            0.167   21         0  
4            2.288   33         1

2) Brief description of data.

This Data set has information about different health conditions that can influence the development of diabetes. Using those conditions I will try to make a Machine Learning Model that can predict diabetes.

In [7]:
diabetes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   NoofPregnancies    768 non-null    int64  
 1   GlucoseLevel       768 non-null    int64  
 2   DiastolicPressure  768 non-null    int64  
 3   Insulin Level      768 non-null    int64  
 4   BMI                768 non-null    float64
 5   RelativeHistory    768 non-null    float64
 6   Age                768 non-null    int64  
 7   Diabetic           768 non-null    int64  
dtypes: float64(2), int64(6)
memory usage: 48.1 KB


3) What is the target?

My target column will be - 'Diabetic' 

4) Is this a classification or regression problem?

It is a classification problem. I decided to choose a set for classification, because I used regression in my first Project.

5) How many features? 

This Data has 8 columns.

6) How many rows of data?

This Data has 768 rows.

In [8]:
diabetes_df.shape

(768, 8)

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

I will need to read more about all health conditions, to be able to fill missing Data the best possible way.